packages/core-ethereum/src/ethereum.ts

```diff
@@ -91,12 +97,22 @@ export async function createChainWrapper(providerURI: string, privateKey: Uint8A
     return response
   }
 
+  /**
+   * Update nonce-tracker and transaction-manager, broadcast the transaction on chain, and listen
+   * to the response until reaching block confirmation.
+   * @param checkDuplicate If the flag is true (default), check if an unconfirmed (pending/mined) transaction with the same payload has been sent
+   * @param method contract method
+   * @param rest contract arguments
+   * @returns Promise of a ContractTransaction
+   */
   async function sendTransaction<T extends (...args: any) => Promise<ContractTransaction>>(
+    checkDuplicate: Boolean,
+    populatedTx: UnsignedTransaction,
     method: T,
     ...rest: Parameters<T>
```

Comment:
could we simplify this somehow? for example, do we need to pass `populatedTx`? can't this be derived from calling `method(...rest)`?